build.rs

```diff
@@ -0,0 +1,17 @@
+fn main() {
+    let cfg = match autocfg::AutoCfg::new() {
+        Ok(cfg) => cfg,
+        Err(e) => {
+            println!("cargo:warning=async-io: failed to detect compiler features: {e}");
+            return;
+        }
+    };
+
+    // emit expected config settings; may be replaced with autocfg::emit_possibility
+    // if we update to autocfg 1.5
+    println!("cargo:rustc-check-cfg=cfg(has_std__io__PipeWriter)");
+    println!("cargo:rustc-check-cfg=cfg(has_std__io__PipeReader)");
+
+    cfg.emit_has_type("std::io::PipeReader");
```

Comment:
Could you use a build script in the same format as https://github.com/smol-rs/polling/pull/39's one? Specifically:

- Calling rustc in a build script to compile code (which emit_has_type does) is relatively fragile (does not work in some host environments and does not work for tier 3 targets). So I would recommend checking the version.
- I would prefer to invert cfg because [it is also works well with non-cargo build systems](https://github.com/smol-rs/polling/pull/39#issuecomment-1237278819).